app/code/community/Hps/Transit/Model/Source/CvvResultCodes.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * @category   Hps
+ * @package    Hps_Transit
+ * @copyright  Copyright (c) 2015 Heartland Payment Systems (https://www.magento.com)
+ * @license    https://github.com/hps/transit-magento-extension/blob/master/LICENSE  Custom License
+ */
+
+class Hps_Transit_Model_Source_CvvResultCodes
+{
+    public function toOptionArray()
+    {
+        return array(
+            array(
+                'value' => 'D2020',
+                'label' => 'CVV2 verification failed'
+            ),
+            array(
+                'value' => 'D2027',
+                'label' => 'AVS and CVV2 failed'
+            )
+        );
+    }
```

Comment:
We should be using the individual CVV result codes instead of the gateway's general response code.

Reply:
ok.. updated the result codes